storages/clickhouse/clickhouse.go

```diff
@@ -387,36 +341,80 @@ func (ch *clickHouse) Read(ctx context.Context, queries []base.Query) (*prompb.R
 		}
 	}
 
+
+
 	res := &prompb.ReadResponse{
 		Results: make([]*prompb.QueryResult, len(queries)),
 	}
+	global := time.Now()
+
 	for i, q := range queries {
+		label := fmt.Sprintf("%s", q.Matchers)
+		label = strings.TrimSuffix(label, "}")
+		label = strings.TrimPrefix(label, "{")
+		var replacer = strings.NewReplacer(".*", "%")
+		label = replacer.Replace(label)
+		var fingerprintQuery = "SELECT DISTINCT fingerprint, labels from prometheus.time_series WHERE labels "
+		slicedLabels := strings.Split(label, ",")
+
+		for _, slice := range slicedLabels {
+			ch.l.Debugf("Slice: %s", slice)
+			if strings.Contains(slice, "=~\"") {
```

Comment:
add few comments why you are doing this for future

Reply:
Done.